### PR TITLE
Add smite reasons

### PR DESCRIPTION
**Pull request by ImScoop**
Back from my old issues, Here is my PR for it.

### CODE REVIEW

**Comment by Wild1145** [comment]
I think (if I remember) that @JeromSar has always wanted reasons to be optional, I personally think they should be a requirement, but if I remember correctly I think that it was raised before to be an optional setting.

**Comment by Scuph** [comment]
@Wild115 So the make the minimum args length one?

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 1]
```diff
@@ -4,19 +4,21 @@
...
 @CommandPermissions(level = AdminLevel.SUPER, source = SourceType.BOTH)
-@CommandParameters(description = "Someone being a little bitch? Smite them down...", usage = "/<command> [playername]")
+@CommandParameters(description = "Someone being a little bitch? Smite them down...", usage = "/<command> [playername] [reason]")
 public class Command_smite extends TFM_Command
 {
     @Override
     public boolean run(CommandSender sender, Player sender_p, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
     {
-        if (args.length != 1)
+        if (args.length != 2)
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Reason should be optional.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 2]
```diff
@@ -33,10 +35,12 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
 
         return true;
     }
+    
+    String smiteReason = StringUtils.join(ArrayUtils.subarray(args, 1, args.length), " ");
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: StringUtils.join can do this without requiring ArrayUtils. Check the APIdocs

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 3]
```diff
@@ -33,10 +35,12 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
 
     public static void smite(final Player player)
     {
-        TFM_Util.bcastMsg(player.getName() + " has been a naughty, naughty boy.", ChatColor.RED);
+        TFM_Util.bcastMsg(player.getName() + " has been a naughty, naughty boy.\nThey have thus been smitten/nReason - " + smiteReason, ChatColor.RED);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: @Scuph Broadcasts shouldn't have newlines. Additionally "They have thus been smitten" is unnecessary. If there's a reason present, it should be on the same line.

**Comment by Scuph** [comment]
I fixed some things, but the stringutils I will look into

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 4]
```diff
@@ -33,10 +35,12 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
 
         return true;
     }
+    
+    String smiteReason = StringUtils.join(args, 1, args.length), " ");
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: This doesn't work if there's no reason. Test your code!!

**Comment by Scuph** [comment]
@Jeromsar I said id look later

**Comment by Scuph** [comment]
@JeromSar done.

**Comment by Scuph** [comment]
@JeromSar You keep marking Inactivity on my PRs, while this one was tested and works, please review what you are doing.

**Comment by Adaptified** [comment]
Still sitting here :)

**Comment by Adaptified** [comment]
@JeromSar Tried a more simpler method.

**Comment by JeromSar** [comment]
@Adaptified There's still some issues with this PR:
* Reasons aren't optional
* Nonstandard usage format
* It does not compile, let alone work.

**Pull request closed without merging**